templates/observation.html

```diff
@@ -414,6 +414,24 @@ <h5 class="modal-title" id="confirmModalLabel">Reset media</h5>
 
   <script>
     $(document).ready(function () {
+      var formChanged = false;
+
+      $('#form :input').on('change input', function() {
+        formChanged = true;
+      });
+
+      $('.nav-link').click(function() {
```

Comment:
Can't you use `beforeunload` event instead?

Reply:
I was thinking about this but I didn't want to use it in every case (like update draft button) and I'm not sure if I can in function exclude those.

Reply:
As I remember you can check event target to do that, so you can save one listener with such approch.

Reply:
I tried that but it was always set to document.

Reply:
so ok - we can do this as it is.